lib/src/database/database_file_storage_io.dart

```diff
@@ -33,6 +33,20 @@ mixin DatabaseFileStorage {
     return null;
   }
 
+  Future<bool> deleteFile(Uri mxcUri) async {
+    final fileStorageLocation = this.fileStorageLocation;
+    if (!supportsFileStoring || fileStorageLocation == null) return false;
+
+    final dir = Directory.fromUri(fileStorageLocation);
+
+    final file = File('${dir.path}/${mxcUri.toString().split('/').last}');
```

Comment:
We repeat this code a lot now. Can we somehow put that in some shared code and add some validation, that the last part isn't `..` or similar? Also, because we don't use the full namespacing, this can clobber files from other servers if the server is malicious, maybe we want to fix that as well? :)

Reply:
Done :)